homeassistant/components/reolink/sensor.py

```diff
@@ -76,6 +76,37 @@ class ReolinkHostSensorEntityDescription(
     ),
 )
 
+HDD_SENSORS = (
+    ReolinkSensorEntityDescription(
+        key="hdd_storage",
+        cmd_key="GetHddInfo",
+        translation_key="hdd_storage",
+        icon="mdi:harddisk",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        value=lambda api, idx: api.hdd_storage(idx),
+        supported=lambda api, idx: (
+            api.supported(None, "hdd") and api.hdd_type(idx) == "HDD"
+        ),
+    ),
+    ReolinkSensorEntityDescription(
+        key="sd_storage",
+        cmd_key="GetHddInfo",
+        translation_key="sd_storage",
+        icon="mdi:micro-sd",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        value=lambda api, idx: api.hdd_storage(idx),
```

Comment:
As `cmd_key` and `value` fn are the same for both entities, I wonder if the camera/device can only have either HDDs or SDs. If that is the case, we could merge them and set the translation key not in the entity description but directly in the entity (as we currently do for the placeholders).

Reply:
You are right, I will merge them.